cmd/maya-exporter/app/collector/pool/pool_test.go

```diff
@@ -1,64 +1,26 @@
 package pool
 
 import (
-	"errors"
 	"fmt"
-	"io"
-	"io/ioutil"
-	"net/http"
-	"net/http/httptest"
 	"regexp"
-	"strconv"
-	"strings"
-	"sync"
 	"testing"
-	"time"
 
-	"github.com/prometheus/client_golang/prometheus"
-	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/openebs/maya/cmd/maya-exporter/app/collector/mock"
+	"github.com/openebs/maya/pkg/util"
 )
 
-var count int
-
-type testRunner struct {
-	stdout  []byte
-	isError bool
-}
-
-func (r testRunner) RunCombinedOutput(cmd string, args ...string) ([]byte, error) {
-	return nil, nil
-}
-
-func (r testRunner) RunStdoutPipe(cmd string, args ...string) ([]byte, error) {
-	return nil, nil
-}
-
-func (r testRunner) RunCommandWithTimeoutContext(timeout time.Duration, cmd string, args ...string) ([]byte, error) {
-	if r.isError {
-		switch count {
-		case 1:
-			count++
-			return []byte("no pools available"), nil
-		case 2:
-			return []byte("ONLINE"), nil
-		case 3:
-			count = 1
-			return nil, errors.New("some dummy error")
-		default:
-			return nil, errors.New("some dummy error")
-		}
-	}
-	return r.stdout, nil
-}
-
-func TestGetZpoolStats(t *testing.T) {
+func TestPoolCollector(t *testing.T) {
 	cases := map[string]struct {
-		run            testRunner
-		match, unmatch []*regexp.Regexp
+		// run, mock the behaviour of zpool binary
+		run util.MockRunner
```

Comment:
It will be good if mock structure can stick to use of primitive datatypes
This helps in understanding the test logic better.
e.g.
```go
tests := map[string]struct {
  zpoolOutput string
  expectedPoolSize string
  expectedUsedPoolCapacityPercent string
  // ...
}{}
```
Use of MockRunner, regexp, etc should be delegated to building the test logic or in some helper functions in this file.

Reply:
@AmitKumarDas I am sticking to older approach, which is quite simpler, above approach will require lot more work because i need to write a logic to generate regular expression based on the above inputs.